rules-reviewed/camel3/camel2/xml-removed-components.windup.groovy

```diff
@@ -0,0 +1,82 @@
+package camel3.camel2
+
+import org.apache.commons.lang3.StringUtils
+import org.jboss.windup.config.GraphRewrite
+import org.jboss.windup.config.metadata.TechnologyReference
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation
+import org.jboss.windup.config.query.Query
+import org.jboss.windup.config.query.QueryPropertyComparisonType
+import org.jboss.windup.graph.model.resource.FileModel
+import org.jboss.windup.reporting.category.IssueCategory
+import org.jboss.windup.reporting.category.IssueCategoryRegistry
+import org.jboss.windup.reporting.config.Hint
+import org.jboss.windup.reporting.config.Link
+import org.jboss.windup.rules.apps.xml.condition.XmlFile
+import org.jboss.windup.rules.apps.xml.model.XmlFileModel
+import org.jboss.windup.rules.apps.xml.model.XmlTypeReferenceModel
+import org.jboss.windup.rules.files.condition.FileContent
+import org.ocpsoft.rewrite.config.Condition
+import org.ocpsoft.rewrite.context.EvaluationContext
+
+import java.util.function.BiFunction
+import java.util.function.Function
+
+final String FROM_XML_FILES_IN_PROJECT = "xmlFilesInProject"
+final String FROM_FILES_IN_PROJECT = "filesInProject"
+final Set<String> componentsAdded = ["aws-cw", "aws-ddb", "aws-ec2", "aws-iam", "aws-kinesis", "aws-kms", "aws-lambda", "aws-mq", "aws-s3", "aws-sdb", "aws-ses", "aws-sns", "aws-sqs", "aws-swf", "aws-xray"]
+
+final IssueCategory mandatoryIssueCategory = new IssueCategoryRegistry().getByID(IssueCategoryRegistry.MANDATORY)
+final Link awsLink = Link.to("Camel 3 - Migration Guide: AWS", "https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_aws")
+
+final BiFunction<String, String, Boolean> isThereArtifactId = { String xmlDependenciesBlock, String component -> 
+    xmlDependenciesBlock.contains("<artifactId>camel-$component</artifactId>" as CharSequence) }
+
+final BiFunction<String, String, Condition> xmlCondition = { String component, String namespace -> 
+    XmlFile.from(FROM_XML_FILES_IN_PROJECT)
+            .matchesXpath("//*/c:route/*[starts-with(@uri, '$component:')]")
```

Comment:
If I understand correctly this is filtering those nodes in the XML containing the uri attribute equals to '$component:`.  
This filter would cover this code ? 
`<pollEnrich id="pollEnrich">  
    <simple>aws-s3:{{S3_BUCKET}}?amazonS3Client=#s3client&amp;deleteAfterRead=false&amp;fileName=${body.payloadStorageId}</simple>  
</pollEnrich>`

Reply:
`the uri attribute equals to $component` The condition is not `equals` but `starts-with` so it will match your example.

Reply:
So you are saying that the node Simple node has an attribute called URI. If you pay attention in that case we have a node called `simple` which `text` starts with `aws-s3`.  
Simple is totally different than a component with this usage : <to uri="log:INFO?showBody=true&amp;showHeaders=true" />

Reply:
@jonathanvila there's no `uri` attribute in your example about `simple` so the condition won't match

Reply:
@mrizzi exactly , that's what I mean.

Reply:
Due to an initial formatting issue, there were some misunderstandings on the request here: it was not intended to have an agreement on how the current condition works :sweat_smile: 
To summarize it effectively: 
**can the filter be enhanced covering also the `<simple>` use case reported?**

@johnpoth and @mmelko i will fill (reopen) a Jira for this.

Is this a good enough solution for you @jonathanvila related to this PR?

Reply:
+1

Reply:
Reported in https://issues.redhat.com/browse/WINDUPRULE-593